File: 2103-rings-and-rods/2103-rings-and-rods.java
class Solution {
    public int countPoints(String rings) {
        int[] rods = new int[10];
        for(int i = 0; i < rings.length() - 1; i += 2){
            if(rings.charAt(i) == 'R')
                rods[rings.charAt(i + 1) - '0'] |= (1 << 0);
            if(rings.charAt(i) == 'G')
                rods[rings.charAt(i + 1) - '0'] |= (1 << 1);
            if(rings.charAt(i) == 'B')
                rods[rings.charAt(i + 1) - '0'] |= (1 << 2);
        }
        int total = 0;
        for(int i = 0; i < rods.length; i++){
            if(rods[i] == 7)
                total++;
        }
        return total;
    }
}

File: 2103-rings-and-rods/2103-rings-and-rods.cpp
class Solution {
public:
    int countPoints(string rings) {
    //     int rods[10] = {};
    // for (int i = 0; i < rings.size(); i += 2) 
    // {
    //     int color = rings[i] == 'R' ? 1 : rings[i] == 'G' ? 2 : 4;
    //     rods[rings[i + 1] - '0'] |= color;
    // }
    // return count(begin(rods), end(rods), 7);
        int c[30]={};
        for(int i=0;i<rings.size();i+=2)
        {
           if(rings[i]=='R')c[rings[i+1]-'0']=1;
           else if(rings[i]=='B')c[rings[i+1]-'0'+10]=1;
            else c[rings[i+1]-'0'+20]=1;
        }
        int res=0;
        for(int i=0;i<=9;i++)
            res+=(c[i] && c[i+10] && c[i+20]);
        return res;
    }
};